File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.java
import java.util.*;
class Solution {
    public int solution(int[] elements) {
        int answer = 0;
        HashSet<Integer> h = new HashSet<>();
        int step = 1;
        
        while(step != elements.length + 1){
            for(int i=0;i<elements.length;i++){
                int sum = 0;
                for(int j=i;j<i+step;j++){
                    sum += elements[j % elements.length];
                }
                h.add(sum);
            }
            step++;
        }
        
        return h.size();
    }
}

File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.cpp
#include <bits/stdc++.h>

using namespace std;

map<int,int> m;

void recur(int cnt, int idx, vector<int> elements) {
    int result = 0;
    
    for (int i = 0; i<cnt;i++) {
        result += elements[(idx + i) % elements.size()];
    }
    
    if (m[result] == 0) m[result]++;
}

int solution(vector<int> elements) {
    int answer = 0;
    
    for (int i=1;i<=elements.size();i++) {
        for (int j=0;j<elements.size();j++) {
            recur(i,j, elements);
        }
    }
    
    return m.size();
}